.github/workflows/android.yml

```diff
@@ -1,32 +1,58 @@
 name: Build Android and upload to Play Store
 on: 
-  push: 
-    branches: [ main,pre-main ]
+  workflow_call:
+    secrets:
+      CODESIGNING_DEPLOY_KEY:
+        required: true
+      GOOGLE_PLAY_JSON_KEY:
+        required: true
+      MATCH_PASSWORD:
+        required: true
+
 jobs:
-  build-ios: 
+
+  build-droid: 
     name: Build Flutter Android
     runs-on: ubuntu-20.04
     steps:
+      - name: Set environment variables
+        run: |
+          if [[ "${{ github.ref }}" == "refs/heads/main" ]]; then
+            echo "Running on the main branch, deploying to production."
+            echo "export package_name=net.givtapp.kids" >> $GITHUB_ENV
+            echo "export flavor=production" >> $GITHUB_ENV
+            echo "export track=production" >> $GITHUB_ENV
+          else
+            echo "Running on a different branch, deploying to test."
+            echo "export package_name=net.givtapp.kids.test" >> $GITHUB_ENV
+            echo "export flavor=development" >> $GITHUB_ENV
+            echo "export track=internal" >> $GITHUB_ENV
+          fi
+
```

Comment:
I see that you moved everything from the fast file into the .yaml

Reply:
I figured it would allow for branching strategy logic to be controlled in GHA for more oversight, and build related things are kept in the fastfile to keep it lean and easily parsed, spread them by subject.